source/common/http/filter_chain_helper.cc

```diff
@@ -76,7 +76,7 @@ FilterChainUtility::createSingletonDownstreamFilterConfigProviderManager(
   std::shared_ptr<Http::DownstreamFilterConfigProviderManager>
       downstream_filter_config_provider_manager =
           context.singletonManager().getTyped<Http::DownstreamFilterConfigProviderManager>(
-              SINGLETON_MANAGER_REGISTERED_NAME(upstream_filter_config_provider_manager),
+              SINGLETON_MANAGER_REGISTERED_NAME(downstream_filter_config_provider_manager),
```

Comment:
Is this a bugfix?

Reply:
yep.  It wasn't a problem before because because there were no pure upstream filters - everything was accidentally using the downstream factory.